### PR TITLE
Use use-wallet

### LINKED ISSUE #6: Use useWallet()

### PR DESCRIPTION
**Pull request by andy-hook**
Resolves #6 

I wanted to understand the `use-wallet` api so implemented it here.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ export function useNow(updateEvery = 1000) {
 }
 
 export function usePostEmail() {
-  const { account } = useWeb3Connect() || ''
+  const { account } = useWallet() || ''
```
Comment (andy-hook) [inline lib/utils.js:89]: I didn't want to run the event logging code in tests so grabbed the account directly.

File: lib/wallet.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useContext, useMemo } from 'react'
+import env from './environment'
```
Comment (sohkai) [inline lib/wallet.js:2]: Minor style nitpick: we generally order imports to make it clearer which imports are "controlled" by us:

1. React-related (`react`, `react-dom`, `styled-components`, etc)
1. External dependencies we don't control
1. External dependencies we do control
1. Internal modules

So to apply in this case, we would reorder to:

1. React
1. `ethers`
1. `use-wallet`
1. Internal modules

File: lib/wallet.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import { UnsupportedChainError, UseWalletProvider, useWallet } from 'use-wallet'
+import { providers } from 'ethers'
+import { getNetworkName } from './web3-utils'
+import { trackEvent } from 'lib/analytics'
+
+const CHAIN_ID = Number(env('CHAIN_ID'))
+const WalletAugmentedContext = React.createContext()
+
+function logError(err, ...messages) {
+  if (typeof window !== 'undefined') {
```
Comment (sohkai) [inline lib/wallet.js:12]: Just wondering, do we need to do this for SSR?

File: lib/utils.js (hunk 1 above)
Comment (sohkai) [reply lib/utils.js:89]: Ah, the analytics code should be set up to skipped during non `NODE_ENV=production` environments; I might've forgot to do this when initially setting up the analytics 🤭

File: lib/wallet.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useContext, useMemo } from 'react'
+import env from './environment'
+import { UnsupportedChainError, UseWalletProvider, useWallet } from 'use-wallet'
+import { providers } from 'ethers'
+import { getNetworkName } from './web3-utils'
+import { trackEvent } from 'lib/analytics'
+
+const CHAIN_ID = Number(env('CHAIN_ID'))
+const WalletAugmentedContext = React.createContext()
```
Comment (Evalir) [inline lib/wallet.js:9]: This is a bit of a nitpick, but I think we can leave a line in between these, as they're not related.

File: lib/wallet.js (hunk 3 above)
Comment (andy-hook) [reply lib/wallet.js:12]: Yup

**Comment by andy-hook** [comment]
> LGTM!
> 
> Just to double check, did we try connecting on mainnet and Rinkeby with the various wallets? I think we should be fine—`ethers` seems to be more robust in this case than `web3` (which we have various problems with in the client).

I did, was a good excuse to try out all of the supported wallets :)

File: lib/wallet.js (hunk 3 above)
Comment (sohkai) [reply lib/wallet.js:12]: Cool—can we leave a comment? How does this even get called when in SSR?

Comment (andy-hook) [reply lib/wallet.js:12]: Wasn't my original code but AFAIK it'll try to access `alert` on the `window` object and bark when it's undefined. I'll leave a comment.

File: lib/utils.js [hunk 5]
```diff
@@ -85,45 +82,6 @@ export function useNow(updateEvery = 1000) {
   return now
 }
 
-export function usePostEmail() {
```
Comment (sohkai) [inline lib/utils.js:88]: @Evalir Are we sure we don't use this anymore? Isn't this how users register their email if they're first time jurors?

Comment (Evalir) [reply lib/utils.js:88]: Yep; remember this is not convert.aragon.org :) Here we don't ask for emails as we're not activating their balance. I remember asking if we would include this in the site, but decided not to at some point.

Comment (andy-hook) [reply lib/utils.js:88]: > Yep; remember this is not convert.aragon.org :) Here we don't ask for emails as we're not activating their balance. I remember asking if we would include this in the site, but decided not to at some point.

You meant anj.aragon.org right? so we're good here? :)

Comment (Evalir) [reply lib/utils.js:88]: > You meant anj.aragon.org right? so we're good here? :)
Haha indeed!

Comment (sohkai) [reply lib/utils.js:88]: Good catch, yes this makes sense!

**Pull request merged**